src/lib.rs

```diff
@@ -230,7 +232,24 @@ fn setup_bitcoind(
         .bitcoind_config
         .as_ref()
         .ok_or(StartupError::MissingBitcoindConfig)?;
-    let bitcoind = BitcoinD::new(bitcoind_config, wo_path_str)?;
+
+    let is_remote = bitcoind_config.addr.ip() != IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
```

Comment:
This detection seems brittle. Can't we just query the system to know whether something is running at the configured (ip, port)?